WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -802,4 +802,48 @@
       expect(competition.registration_full?).to be true
     end
   end
+
+  context "when changing the competition's date" do
+    let(:competition) {
+      FactoryBot.create :competition,
+                        with_schedule: true,
+                        start_date: Date.parse("2018-10-24"),
+                        end_date: Date.parse("2018-10-26")
+    }
+    let(:all_activities) {
+      competition.competition_venues.includes(venue_rooms: { schedule_activities: [:child_activities] }).map(&:all_activities).flatten
+    }
+
+    def change_and_check_activities(new_start_date, new_end_date)
+      on_first_day, on_last_day = all_activities.partition { |a| a.start_time.to_date == competition.start_date }
+      # the factory define one activity per day, the two lines below are
+      # basically safe guards against a future change to the competition's factory.
+      expect(on_first_day).not_to be_empty
+      expect(on_last_day).not_to be_empty
+      competition.update(start_date: new_start_date,
+                         end_date: new_end_date)
+      on_first_day.map(&:reload)
+      on_last_day.map(&:reload)
```

Comment:
Perhaps `all_activities.each(&:reload)` would do the job, as all the arrays refer to the same objects.

Reply:
Right thanks!